applications/tari_base_node/src/main.rs

```diff
@@ -174,6 +176,36 @@ fn main_inner() -> Result<(), ExitCodes> {
         return Ok(());
     }
 
+    // create recovery db
+    let temp_db = if bootstrap.rebuild_db {
+        info!(
+            target: LOG_TARGET,
+            "Node is in recovery mode, it will try to rebuild the db"
+        );
+        match &node_config.db_type {
+            DatabaseType::LMDB(p) => {
+                let _backend = create_recovery_lmdb_database(&p).map_err(|err| {
+                    error!(target: LOG_TARGET, "{}", err);
+                    ExitCodes::UnknownError
+                })?;
+                let new_path = Path::new(p).join("temp_recovery");
+                Some(
+                    create_lmdb_database(&new_path, node_config.db_config.clone(), MmrCacheConfig::default()).map_err(
+                        |e| {
+                            error!(target: LOG_TARGET, "Error opening recovery db: {}", e);
+                            ExitCodes::UnknownError
+                        },
+                    )?,
+                )
+            },
+            _ => {
+                error!(target: LOG_TARGET, "Recovery mode is only available for LMDB");
+                return Ok(());
+            },
+        }
+    } else {
+        None
+    };
```

Comment:
This should all go into a separate module.

Recovery could be a completely separate executable, so it doesn't go it the main node event loop.